pkg/scheduler/core/generic_scheduler.go

```diff
@@ -77,28 +101,83 @@ func (g *genericScheduler) Schedule(ctx context.Context, placement *policyv1alph
 	return result, nil
 }
 
+// calcReservedClusters returns clusters that should be reserved in `binding.spec.clusters`
+func calcReservedClusters(
+	bindClusters, readyClusters, unreadyClusters sets.String,
+	candidateClusters []*clusterv1alpha1.Cluster,
+	placement *policyv1alpha1.Placement) sets.String {
+	// reservedClusters are ready clusters in `binding.spec.clusters`
+	reservedClusters := bindClusters.Intersection(readyClusters)
+	// unreadyBindClusters are "unready" clusters in `binding.spec.clusters`
+	unreadyBindClusters := bindClusters.Intersection(unreadyClusters)
+
+	if unreadyBindClusters.Len() > 0 {
+		if !features.FeatureGate.Enabled(features.Failover) {
+			// reserve unready clusters if failover is disabled
+			reservedClusters = reservedClusters.Union(unreadyBindClusters)
+		} else if len(candidateClusters) < unreadyBindClusters.Len() {
+			// *DON'T evict unready clusters in `binding.spec.clusters` if there are insufficient candidate clusters*
```

Comment:
I think the num of bindClusters isn't have to be the cluster num of the next scheduling